src/util/rocksdb/mod.rs

```diff
@@ -360,12 +361,76 @@ pub fn compact_range(
     db.compact_range_cf_opt(handle, &compact_opts, start_key, end_key);
 }
 
+/// Prepare the SST file for ingestion.
+/// The purpose is to make the ingestion retryable when using the `move_files` option.
+/// Things we need to consider here:
+/// 1. We need to access the original file on retry, so we should make a clone
+///    before ingestion.
+/// 2. `RocksDB` will modified the global seqno of the ingested file, so we need
+///    to modified the global seqno back to 0 so that we can pass the checksum
+///    validation.
+/// 3. If the file has been ingested to `RocksDB`, we should not modified the
+///    global seqno directly, because that may corrupt RocksDB's data.
+#[cfg(target_os = "linux")]
```

Comment:
seem mac and windows also support hardlink

Reply:
Yep, but we can't get the hard link count on mac and windows.